File: C++/30.前端控制器模式/FrontControllerPatternDemo.cpp
/**
 * @file FrontControllerPatternDemo.cpp
 * @author {Layne} ({shu_huanglei@163.com})
 * @brief
 * @version 0.1
 * @date 2019-12-06
 *
 * @copyright Copyright (c) 2019
 *
 */

#include "FrontController.h"

int main(int argc, char const *argv[]) {
    std::shared_ptr<FrontController> front_controller = std::make_shared<FrontController>();
    front_controller->DispatchRequest("HOME");
    front_controller->DispatchRequest("STUDENT");
    return 0;
}

File: Objective-C/30.FrontControllerPattern/FrontControllerPatternDemo.m
//
//  FrontControllerPatternDemo.m
//  30.FrontControllerPattern
//
//  Created by Layne on 2020/11/22.
//

#import "FrontController.h"

int main(int argc, const char* argv[]) {
    @autoreleasepool {
        FrontController* frontController = [[FrontController alloc] init];
        [frontController dispatchRequest:@"HOME"];
        [frontController dispatchRequest:@"STUDENT"];
    }
    return 0;
}
